Modules/_xxtestfuzz/fuzzer.c

```diff
@@ -501,6 +501,60 @@ static int fuzz_elementtree_parsewhole(const char* data, size_t size) {
     return 0;
 }
 
+#define MAX_PYCOMPILE_TEST_SIZE 16384
+static char pycompile_scratch[MAX_PYCOMPILE_TEST_SIZE];
+
+static const int start_vals[] = {Py_eval_input, Py_single_input, Py_file_input};
+const size_t NUM_START_VALS = sizeof(start_vals) / sizeof(start_vals[0]);
+
+static const int optimize_vals[] = {-1, 0, 1, 2};
+const size_t NUM_OPTIMIZE_VALS = sizeof(optimize_vals) / sizeof(optimize_vals[0]);
+
+/* Fuzz `PyCompileStringExFlags` using a variety of input parameters.
+ * That function is essentially behind the `compile` builtin */
+static int fuzz_pycompile(const char* data, size_t size) {
+    if (size > sizeof(pycompile_scratch)) {
+        return 0;
+    }
+
+    // Need 2 bytes for parameter selection plus 1 for null terminator
+    if (size < 2 + 1)
+        return 0;
+
+    // Use first byte to determine element of `start_vals` to use
+    unsigned char start_idx = (unsigned char) data[0];
+    int start = start_vals[start_idx % NUM_START_VALS];
+
+    // Use second byte to determine element of `optimize_vals` to use
+    unsigned char optimize_idx = (unsigned char) data[1];
+    int optimize = optimize_vals[optimize_idx % NUM_OPTIMIZE_VALS];
+
+    // Create a null-terminated C string from the remaining input
+    memcpy(pycompile_scratch, data + 2, size - 2);
```

Comment:
Shouldn't this be `size - 1`, otherwise aren't we overwriting data from the fuzzer?

Reply:
The first 2 bytes of the fuzzer's input are being used to choose the value of `start_vals` and `optimize_vals` (see lines 524-530). We don't want to treat those 2 bytes as Python source code input in the first argument of `Py_CompileStringExFlags`, hence the `data + 2` and `size - 2` here.

Yes, the final byte of fuzzer input gets overwritten with a `nul` terminator.

Reply:
Okay, this should be fixed now, with none of the fuzzer's input getting overwritten.